drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -384,7 +384,7 @@ static int _set(netdev_t *netdev, netopt_t opt, void *val, size_t len)
             /* don't set res to set netdev_ieee802154_t::pan */
             break;
         case NETOPT_CHANNEL:
-            assert(len != sizeof(uint8_t));
+            assert(len <= sizeof(uint16_t));
```

Comment:
Because if the user provides a 16-bit channel (or the shell_commands module for that matter) the cast in `netdev_ieee802154` will cast a `uint16_t` pointer to a `uint8_t` pointer, discarding the least-significant byte, which on a big-endian platform would contain the actual value, so the value would always resolve to 0.